rkt/api_service.go

```diff
@@ -114,16 +114,9 @@ func isPartOf(keyword, s string) bool {
 	return strings.Contains(s, keyword)
 }
 
-// satisfiesPodFilter returns true if the pod satisfies the filter or the filter
-// is nil, which means the pod will be returned.
-// It returns false if the pod doesn't satisfy the filter, which means it will
-// not be returned.
+// satisfiesPodFilter returns true if the pod satisfies the filter.
+// The pod, manifest, filter must not be nil.
```

Comment:
Well, I prefer to pass pointers when the struct is big. But I did a benchmark for this particular function, it doesn't seems memory copy are fast these days, so not a big deal (30ns/op vs 10ns/op).

Like you said, passing the values not necessary solves all problems, especially for structs that have something like `[]*app`. I just feel raw about not being consistent, but I tend to agree to pass the whole struct when possible.